lib/src/git_backend.rs

```diff
@@ -571,7 +590,11 @@ fn deserialize_extras(commit: &mut Commit, bytes: &[u8]) {
                 .iter()
                 .map(|id_bytes| TreeId::from_bytes(id_bytes))
                 .collect();
-            commit.root_tree = MergedTreeId::Merge(merge_builder.build());
+            let merge = merge_builder.build();
+            if let MergedTreeId::Merge(existing_merge) = &commit.root_tree {
+                assert!(existing_merge.is_resolved() || *existing_merge == merge);
```

Comment:
nit: can you add an inline comment or assertion message that we're testing redundant data? It will help future readers.